huntsman/utils/config.py

```diff
@@ -144,3 +146,47 @@ def query_config_server(key=None, name='config_server'):
             
 #==============================================================================
 
+def load_device_config(key=None, config_files=None, logger=None, **kwargs):
+    '''
+    Load the device config from either the config server or local files.
+    
+    Parameters
+    ----------
+    key:
+        The key used to query the config server. Only used if config_files
+        is None.
+    config_files:
+        List of config file names. If None (default), use the config server.
+        
+    Returns
+    -------
+    dict:
+        The config dictionary.
+    '''
+    if logger is None:
+        logger = DummyLogger()
+        
+    #Load config from local files?
+    if config_files is not None:
+        logger.debug(f'Loading config from local file(s).')
+        try:
+            config = load_config(config_files, **kwargs)
```

Comment:
Don't you need to query by ip here as well? Or at least do a:
```suggestion
            config = load_config(config_files, **kwargs)[key]
```